tests/test_noqa.py

```diff
@@ -0,0 +1,274 @@
+import pytest
+
+from vulture.core import NOQA_REGEXP, _parse_error_codes
+from . import check, v
+
+assert v  # Silence pyflakes.
+
+
+@pytest.mark.parametrize(
+    "line, codes",
+    [
+        ("# noqa", ["all"]),
+        ("## noqa", ["all"]),
+        ("# noqa Hi, go on.", ["all"]),
+        ("# noqa: V101", ["V101"]),
+        ("# noqa: V101, V106", ["V101", "V106"]),
+        ("# NoQA: V101,      V103, \t V104", ["V101", "V103", "V104"]),
+    ],
+)
+def test_noqa_regex_present(line, codes):
+    match = NOQA_REGEXP.search(line)
+    parsed = _parse_error_codes(match)
+    assert parsed == codes
+
+
+@pytest.mark.parametrize(
+    "line",
+    [
+        ("# noqa: 123V"),
+        ("# noqa explaination: V012"),
+        ("# noqa: ,V101"),
+        ("# noqa: #noqa: V102"),
+        ("# noqa: # noqa: V102"),
+    ],
+)
+def test_noqa_regex_no_groups(line):
+    assert NOQA_REGEXP.search(line).groupdict()["codes"] is None
+
+
+@pytest.mark.parametrize(
+    "line",
+    [("#noqa"), ("##noqa"), ("# n o q a"), ("#NOQA"), ("# Hello, noqa")],
+)
+def test_noqa_regex_not_present(line):
+    assert bool(NOQA_REGEXP.search(line)) is False
+
+
+def test_noqa_without_codes(v):
+    v.scan(
+        """\
+import this  # noqa
+
+class Cellar:  # noqa
+    @property
+    def wine(self, grapes): # noqa
+        pass
+
+    def serve(self, quantity=50):  # noqa
+        self.quantity_served = quantity  # noqa
+        return
+        self.pour()  # noqa
+"""
+    )
+    check(v.unused_attrs, [])
+    check(v.unused_classes, [])
+    check(v.unused_funcs, [])
+    check(v.unused_imports, [])
+    check(v.unused_props, [])
+    check(v.unreachable_code, [])
+    check(v.unused_vars, [])
+
+
+def test_noqa_specific_issue_codes(v):
+    v.scan(
+        """\
+import this  # noqa: V104
+
+class Cellar:  # noqa: V102
+    @property
+    def wine(self, grapes):  # noqa: V105, V106
```

Comment:
Same here. let's add a separate unused variable inside of this method. The we can also remove the `pass`.

Reply:
I added the variable - `year`. Just making sure that you in fact, meant this. If yes, can we mark this as resolved?

Reply:
I'd drop the useless return and remove the V106 noqa from the line above, but then it's resolved, yes.

Reply:
Ahh, got it.